# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../ccache/libtool-options.patch
# Copyright (C) 2004 - 2005 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

The -M* options that control dependency file creation do not affact the output
and especially -MF is used with randomized filenames by libtool and thus
invalidate the hash unnecessarily.

  Rene Rebe <rene@exactcode.de>

--- ccache-2.4/ccache.c	2004-09-13 12:38:30.000000000 +0200
+++ ccache-2.4-improved/ccache.c	2004-12-29 22:38:55.984560272 +0100
@@ -291,6 +291,7 @@
 			    strcmp(args->argv[i], "-include") == 0 ||
 			    strcmp(args->argv[i], "-L") == 0 ||
 			    strcmp(args->argv[i], "-D") == 0 ||
+			    strcmp(args->argv[i], "-MF") == 0 ||
 			    strcmp(args->argv[i], "-idirafter") == 0 ||
 			    strcmp(args->argv[i], "-isystem") == 0) {
 				i++;
@@ -300,6 +301,10 @@
 		if (strncmp(args->argv[i], "-I", 2) == 0 ||
 		    strncmp(args->argv[i], "-L", 2) == 0 ||
 		    strncmp(args->argv[i], "-D", 2) == 0 ||
+		    strncmp(args->argv[i], "-MD", 3) == 0 ||
+		    strncmp(args->argv[i], "-MP", 3) == 0 ||
+		    strncmp(args->argv[i], "-MF", 3) == 0 ||
+		    strncmp(args->argv[i], "-Wp", 3) == 0 ||
 		    strncmp(args->argv[i], "-idirafter", 10) == 0 ||
 		    strncmp(args->argv[i], "-isystem", 8) == 0) {
 			continue;
